lib/duration.ts

```diff
@@ -33,16 +33,16 @@ export class Duration implements Temporal.Duration {
     microsecondsParam: Params['constructor'][8] = 0,
     nanosecondsParam: Params['constructor'][9] = 0
   ) {
-    const years = ES.ToIntegerIfIntegral(yearsParam);
-    const months = ES.ToIntegerIfIntegral(monthsParam);
-    const weeks = ES.ToIntegerIfIntegral(weeksParam);
-    const days = ES.ToIntegerIfIntegral(daysParam);
-    const hours = ES.ToIntegerIfIntegral(hoursParam);
-    const minutes = ES.ToIntegerIfIntegral(minutesParam);
-    const seconds = ES.ToIntegerIfIntegral(secondsParam);
-    const milliseconds = ES.ToIntegerIfIntegral(millisecondsParam);
-    const microseconds = ES.ToIntegerIfIntegral(microsecondsParam);
-    const nanoseconds = ES.ToIntegerIfIntegral(nanosecondsParam);
+    const years = yearsParam === undefined ? 0 : ES.ToIntegerIfIntegral(yearsParam);
```

Comment:
Doesn't ToIntegerIfIntegral already return 0 if it is given undefined?

https://github.com/js-temporal/temporal-polyfill/blob/8ba2a0121ff4458eb98b00d360841df37fc0b403/lib/ecmascript.ts#L239-L240

and

https://github.com/js-temporal/temporal-polyfill/blob/8ba2a0121ff4458eb98b00d360841df37fc0b403/lib/ecmascript.ts#L177-L184

lead me to believe so.

Reply:
Nope. You're looking at an out-of-date `ecmascript.ts` above. Upstream changed (https://github.com/tc39/proposal-temporal/commit/b9579ebe33388bf1b68e3011a449cbcd30cd14ed) its implementation of `ToIntegerIfIntegral` to throw on `NaN`, so the `undefined` check is needed here. (And therefore it's what upstream is doing too.)